apps/heureka/src/components/filters/FilterPills.jsx

```diff
@@ -5,17 +5,32 @@
 
 import React from "react"
 import { Pill, Stack } from "@cloudoperators/juno-ui-components"
-import { useActiveFilters, useFilterActions } from "../../hooks/useAppStore"
+import {
+  useIssueMatchesActiveFilters,
+  useServiceActiveFilters,
+  useComponentActiveFilters,
+  useFilterActions,
+} from "../../hooks/useAppStore"
 import { humanizeString } from "../../lib/utils"
 
+const entityHooks = {
+  IssueMatches: useIssueMatchesActiveFilters,
+  Services: useServiceActiveFilters,
+  Components: useComponentActiveFilters,
+}
+
 const FilterPills = ({ entityName }) => {
```

Comment:
I would pass the active filters as props to the component, so you don't need to implement any logic within it. This way, the component remains purely presentational and only renders what it receives. Does that make sense to you?